azure/datalake/store/core.py

```diff
@@ -168,10 +168,10 @@ def du(self, path, total=False, deep=False):
         else:
             return {p['name']: p['length'] for p in files}
 
-    def df(self):
-        """ Resource summary
-        """
-        return self.azure.call('GETCONTENTSUMMARY')['ContentSummary']
+    def df(self, path):
+        """ Resource summary of path """
+        path = AzureDLPath(path).trim()
+        return self.azure.call('GETCONTENTSUMMARY', path.as_posix())['ContentSummary']
```

Comment:
It looks like all `core.py` methods except `ls()` require a path. The optional path was meant for some CLI commands, like `df` (also included already are `ls` and `du`).